### ISSUE TITLE
Quetion regarding y_class variable

### ISSUE THREAD

**Issue by rimitalahiri92** [ssamot/DANN-keras#2]
Since we can't use the class labels of Xt, how are you y_class as target labels for source classifier part of combined model. And while testing why are you using embedding model to predict instead of using source classification model directly???

**Comment by ssamot** [comment]
Hello - the y_t is not used during learning - it's only used for demonstration purposes to see were learning is going. The embedding model is not used to predict y, it's just used to visualise the embeddings.

**Comment by rimitalahiri92** [comment]
```
y_class = np.concatenate([y0, np.zeros_like(y0)])

stats = model.train_on_batch(X_adv, [y_class, y_adversarial_1],
                                     sample_weight=[sample_weights_class, sample_weights_adversarial])


```
In the first line the np.zeros_like(y0) is the dummy part because you are not using the labels from yt. Then when you train for the second line how do you ignore the those dummy labels, by giving sample weights of class as 0??? 
I am not understanding this part only how the dummy labels don't have any effect in train_on_batch.

**Comment by ssamot** [comment]
The dummy labels have weights of 0, so they don't play any role in the gradient calculation.

**Comment by rimitalahiri92** [comment]
Thank you so much for your explanation, I got it now....

**Comment by rebeen** [comment]
could you please help me to use this code for more than two classes and also more features

**Comment by rebeen** [comment]
@rimitalahiri92  did you understand the model I have some questions, I appreciate if you can help me to understand 

thank you

**Comment by rebeen** [comment]
@ssamot  could you please let me know how the adversarial is working? 

I can see that we have two variables called adversarial  which I have updated for my problem for multi-class classification and I used this `    y_adversarial_1 = np.concatenate([np.ones(( batch_size,11 )), np.zeros((batch_size,11)) ])
`
Thank you

**Comment by rebeen** [comment]
could you please let me know what is the role of `sample_weights_adversarial`  
my second question is what is adversarial here?  can you please shortly let me know?

thank you
